---
title: "Gat201 RNA Seq deseq-analys2s 2021-06-18EH"
author: "Liz Hughes"
date: "18/06/2021"
output:
  html_document:
    toc: yes
    toc_depth: 2
  word_document:
    toc: yes
    toc_depth: '2'
  pdf_document:
    toc: yes
    toc_depth: '2'

---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE , warning=FALSE, message=FALSE)
```

# GAT201 RNA Seq analysis_EHughes

To carry out RNA-Seq I will use the Lexogen Quant Seq 3' mRNA-Seq library prep kit FWD. This is designed to generate illumina compatible libraries of sequences close to the 3' end of polyA RNA.

Work flow: Reverse transcription ⇒ Removal of RNA ⇒ Second strand synthesis ⇒ Library amplification ⇒ Sequencing

4 strains: KN99a, KN99 alpha, Gat201 deletion from the Madhani deletion collection (003) and Gat201 disruption from the Bahn collection (004).

2 conditions: RPMI and RPMI+Serum (10%).

2 BioReps and 3 Tech Reps.

A single colony was grown in YPD, 30 degrees, 200rpm for 5 days.

3.5 ml was transferred to 100ml fresh RPMI or RPMI+10% Serum and incubated at 37 degrees, 60rpm.

2ml samples were taken from each rep at time 0hrs, fix and freeze and put at -80 degrees.

At 30 mins, 2hrs and 4hrs 15 ml samples were collected, fix and freeze and put at -80 degrees.

# Load packages used for analysis
```{r Load packages used for analysis, include=FALSE}

library(tidyr)
library(readr)
library(dplyr)
library(readxl)
library(stringr)
library(ggplot2)
library(cowplot)
library(biobroom)
library(DESeq2)
library(pheatmap)
library(RColorBrewer)
library(magrittr)
library(markdown)
library(forcats)


theme_set(
  theme_cowplot(font_size = 12) +
    theme(panel.grid.major = element_line(colour = "grey80", size = 0.5))
)

```


# Define colours for specific samples

```{r}
WTup_colour   = "red"
WTdown_colour = "green"
Gat201up_colour  = "purple"
Gat201down_colour = "blue"
```


# Load count data and remove unwanted parts of column names

```{r Load count data}
counts <- readr::read_tsv("~/Documents/GitHub/CryptoGat201RNASeq_draft/quantseqfwd_EH_050221/counts.txt",
                   comment = "#") %>% 
          dplyr::rename_with(str_remove_all, pattern = "_S[0-9]+_R1_001_aln.bam")

```
# Load sample sheet and format for DESeq2's requirements: readxl::read_excel

Connect sample names to informative data that generated the data.

Extra steps to format for DESeq2. 

Remove all samples not sequenced: 'dplyr::filter'

Remove notes in column: 'dplyr::select'

Set rownames to make format of table appropriate for DESeq: 'magrittr::set_rownames'

```{r Load sample sheet}
samplesheet <- readxl::read_excel("~/Documents/GitHub/CryptoGat201RNASeq_draft/input_experiment/Gat201_samplesheet.xlsx") %>%
              magrittr::set_rownames(.$SampleID)%>%
              dplyr::mutate(GAT201 = forcats::fct_collapse(Strain, Yes = c("a","A"), No = c("B","M")))
```
We now have all the ingredients to prepare our data object in a form that is suitable for analysis, namely:

counts: a table with the read counts
samplesheet: a table with information about the samples

# Differential expression analysis 
## WT (GAT201 = yes) vs Gat201 deletion (GAT201 = no) 0h only
```{r DESeq 0h WT v GAT201}

samplesheet_0h <- dplyr::filter(samplesheet, Time == "0")
                  print(samplesheet_0h)

counts_0h <-   dplyr::select(counts, samplesheet_0h$SampleID) %>%
               magrittr::set_rownames(counts$Geneid)

dds_0h <- DESeqDataSetFromMatrix(countData = counts_0h,
                                 colData = samplesheet_0h,
                                 design = ~ GAT201)%>%
        DESeq()

resultsNames(dds_0h)
results_0h <- results(dds_0h, name="GAT201_No_vs_Yes")
plotMA(results_0h, ylim=c(-5,5))

tresults_0h <- biobroom::tidy.DESeqResults(results_0h)
head(tresults_0h)

tresults_0h %>%  
dplyr::mutate(minus_log10_pvalue = -log10(p.value)) %>%
readr::write_tsv("~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WT_Gat201_DEseq2_genes0h.txt")

de_WTup_0h <- dplyr::filter(tresults_0h, estimate > 0, p.adjusted < 0.05) %>%
              dplyr::arrange(desc(estimate))
de_WTup_0h

readr::write_tsv(de_WTup_0h, "~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WTup_DEseq2_padj0p05_0h.txt")
readr::write_lines(de_WTup_0h$gene, "~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WTup_DEseq2_padj0p05_0h_genesonly.txt")

de_WTdown_0h <- dplyr::filter(tresults_0h, estimate < 0, p.adjusted < 0.05) %>%
                dplyr::arrange(estimate)
de_WTdown_0h

readr::write_tsv(de_WTdown_0h, "~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WTdown_DEseq2_padj0p05_0h.txt")
readr::write_lines(de_WTdown_0h$gene, "~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WTdown_DEseq2_padj0p05_0h_genesonly.txt")

plot_0h<- qplot(x = baseMean, y = estimate, data = tresults_0h, geom = "point")+
               geom_text(aes(label=ifelse(baseMean> 20000, (gene),'')),hjust=0,vjust=0, cex = 1.5, jitter = 0.5)+
               labs(x ="Mean", y= "log2fold change", title = "WT v Gat201 mutants at 0 hrs")
plot_0h

volcano_0h <- 
  ggplot(data = tresults_0h, aes(x=estimate,y=-log10(p.value))) +
         geom_point(colour = "grey50",size = 0.5) +
         geom_point(data = de_WTup_0h, colour = WTup_colour) +
         geom_point(data = de_WTdown_0h, colour = WTdown_colour) +
         geom_text(aes(label=ifelse(estimate < -2 & -log10(p.value)> 5, (gene),'')),hjust=0,vjust=0, cex = 1.5, jitter = 0.5)+
         geom_text(aes(label=ifelse(estimate > 2 , (gene),'')),hjust=0,vjust=0, cex = 1.5, jitter = 0.5)+
         scale_y_continuous(breaks = c(0,5,10,15,20),
                            limits = c(0,20), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        scale_x_continuous(breaks = seq(-6,6,2),
                            limits = c(-7,7), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        labs(x = "log2-fold change",
        title = "WT vs Gat201 mutants at 0h")
volcano_0h

```

```{r save volvano plot oh}
ggsave("~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WTvGat201_0h_volcano.png",
       width = 5,height=5)
```



## WT (GAT201 = yes) vs Gat201 deletion (GAT201 = no) 30m only
```{r DESeq 30m WT v GAT201}
samplesheet_30m <- dplyr::filter(samplesheet, Time == "30")
                  print(samplesheet_30m)

counts_30m <-   dplyr::select(counts, samplesheet_30m$SampleID) %>%
               magrittr::set_rownames(counts$Geneid)

dds_30m <- DESeqDataSetFromMatrix(countData = counts_30m,
                                 colData = samplesheet_30m,
                                 design = ~ GAT201)%>%
        DESeq()

resultsNames(dds_30m)
results_30m <- results(dds_30m, name="GAT201_No_vs_Yes")
plotMA(results_30m, ylim=c(-5,5))

tresults_30m <- biobroom::tidy.DESeqResults(results_30m)
head(tresults_30m)

tresults_30m %>%  
dplyr::mutate(minus_log10_pvalue = -log10(p.value)) %>%
readr::write_tsv("~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WT_Gat201_DEseq2_genes30m.txt")

de_WTup_30m <- dplyr::filter(tresults_30m, estimate > 0, p.adjusted < 0.05) %>%
              dplyr::arrange(desc(estimate))
de_WTup_30m

readr::write_tsv(de_WTup_30m, "~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WTup_DEseq2_padj0p05_30m.txt")
readr::write_lines(de_WTup_30m$gene, "~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WTup_DEseq2_padj0p05_30m_genesonly.txt")

de_WTdown_30m <- dplyr::filter(tresults_30m, estimate < 0, p.adjusted < 0.05) %>%
                dplyr::arrange(estimate)
de_WTdown_30m

readr::write_tsv(de_WTdown_30m, "~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WTdown_DEseq2_padj0p05_30m.txt")
readr::write_lines(de_WTdown_30m$gene, "~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WTdown_DEseq2_padj0p05_30m_genesonly.txt")

plot_30m<- qplot(x = baseMean, y = estimate, data = tresults_30m, geom = "point")+
               geom_text(aes(label=ifelse(baseMean> 20000, (gene),'')),hjust=0,vjust=0, cex = 1.5, jitter = 0.5)+
               labs(x ="Mean", y= "log2fold change", title = "WT v Gat201 mutants at 30 min")
plot_30m

volcano_30m <- 
  ggplot(data = tresults_30m, aes(x=estimate,y=-log10(p.value))) +
         geom_point(colour = "grey50",size = 0.5) +
         geom_point(data = de_WTup_30m, colour = WTup_colour) +
         geom_point(data = de_WTdown_30m, colour = WTdown_colour) +
         geom_text(aes(label=ifelse(estimate < -2 & -log10(p.value)> 5, (gene),'')),hjust=0,vjust=0, cex = 1.5, jitter = 0.5)+
         geom_text(aes(label=ifelse(estimate > 2 , (gene),'')),hjust=0,vjust=0, cex = 1.5, jitter = 0.5)+
         scale_y_continuous(breaks = c(0,10,20,30,40,50,60),
                            limits = c(0,60), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        scale_x_continuous(breaks = seq(-6,6,2),
                            limits = c(-7,7), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        labs(x = "log2-fold change",
        title = "WT vs Gat201 mutants at 30 min")
volcano_30m

```
```{r save volcano plot 30m}
ggsave("~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WTvGat201_30m_volcano.png",
       width = 5,height=5)
```



## WT (GAT201 = yes) vs Gat201 deletion (GAT201 = no) 2h only
```{r DESeq 2h WT v GAT201}
samplesheet_2h <- dplyr::filter(samplesheet, Time == "120")
                  print(samplesheet_2h)

counts_2h <-   dplyr::select(counts, samplesheet_2h$SampleID) %>%
               magrittr::set_rownames(counts$Geneid)

dds_2h <- DESeqDataSetFromMatrix(countData = counts_2h,
                                 colData = samplesheet_2h,
                                 design = ~ GAT201)%>%
        DESeq()

resultsNames(dds_2h)
results_2h <- results(dds_2h, name="GAT201_No_vs_Yes")
plotMA(results_2h, ylim=c(-5,5))

tresults_2h <- biobroom::tidy.DESeqResults(results_2h)
head(tresults_2h)

tresults_2h %>%  
dplyr::mutate(minus_log10_pvalue = -log10(p.value)) %>%
readr::write_tsv("~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WT_Gat201_DEseq2_genes2h.txt")

de_WTup_2h <- dplyr::filter(tresults_2h, estimate > 0, p.adjusted < 0.05) %>%
              dplyr::arrange(desc(estimate))
de_WTup_2h

readr::write_tsv(de_WTup_2h, "~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WTup_DEseq2_padj0p05_2h.txt")
readr::write_lines(de_WTup_2h$gene, "~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WTup_DEseq2_padj0p05_2h_genesonly.txt")

de_WTdown_2h <- dplyr::filter(tresults_2h, estimate < 0, p.adjusted < 0.05) %>%
                dplyr::arrange(estimate)
de_WTdown_2h

readr::write_tsv(de_WTdown_2h, "~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WTdown_DEseq2_padj0p05_2h.txt")
readr::write_lines(de_WTdown_2h$gene, "~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WTdown_DEseq2_padj0p05_02_genesonly.txt")

plot_2h <- qplot(x = baseMean, y = estimate, data = tresults_2h, geom = "point")+
               geom_text(aes(label=ifelse(baseMean> 20000, (gene),'')),hjust=0,vjust=0, cex = 1.5, jitter = 0.5)+
               labs(x ="Mean", y= "log2fold change", title = "WT v Gat201 mutants at 2 hrs")
plot_2h

volcano_2h <- 
  ggplot(data = tresults_2h, aes(x=estimate,y=-log10(p.value))) +
         geom_point(colour = "grey50",size = 0.5) +
         geom_point(data = de_WTup_2h, colour = WTup_colour) +
         geom_point(data = de_WTdown_2h, colour = WTdown_colour) +
         geom_text(aes(label=ifelse(estimate < -2 & -log10(p.value)> 20, (gene),'')),hjust=0,vjust=0, cex = 1.5, jitter = 0.5)+
         geom_text(aes(label=ifelse(estimate > 2 , (gene),'')),hjust=0,vjust=0, cex = 1.5, jitter = 0.5)+
         scale_y_continuous(breaks = c(0,10,20,30,40,50,60,70,80,90,100),
                            limits = c(0,100), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        scale_x_continuous(breaks = seq(-6,6,2),
                            limits = c(-7,7), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        labs(x = "log2-fold change",
        title = "WT vs Gat201 mutants at 0h")
volcano_2h

```
```{r save volcano plot 2h}
ggsave("~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WTvGat201_2h_volcano.png",
       width = 5,height=5)
```


## WT (GAT201 = yes) vs Gat201 deletion (GAT201 = no) 4h only

Filter for 4h time points from samplesheet
Select from counts table only counts with 'x'
Set rownames to make format of table appropriate for DESeq: 'magrittr::set_rownames'

```{r DESeq 4h WT v GAT201}
samplesheet_4h <- dplyr::filter(samplesheet, Time == "240")
                  print(samplesheet_4h)

counts_4h <-   dplyr::select(counts, samplesheet_4h$SampleID) %>%
               magrittr::set_rownames(counts$Geneid)

dds_4h <- DESeqDataSetFromMatrix(countData = counts_4h,
                                 colData = samplesheet_4h,
                                 design = ~ GAT201)%>%
        DESeq()
```

### MA plot 4h
```{r result DESeq 4h WT v GAT201}
resultsNames(dds_4h)
results_4h <- results(dds_4h, name="GAT201_No_vs_Yes")
plotMA(results_4h, ylim=c(-5,5))

```
### Tidy version of results_4h
'tidy.DESeqResults' takes big object and just gives the data frame for results_4h


```{r Tidy version of result DESeq 4h WT v GAT201}
tresults_4h <- biobroom::tidy.DESeqResults(results_4h)
head(tresults_4h)
```

### Write full output to table for analysis with VolcaNoseR for results_4h
minus_log10_pvalue need to calcultae this for volcanoser.
This takes the work done in the script so far and puts it in a sensible format and exports it to a file:
WT_Gat201_DEseq2_genes4h.txt (count analysis and sample sheet to get DE data)

```{r}
tresults_4h %>%  
dplyr::mutate(minus_log10_pvalue = -log10(p.value)) %>%
readr::write_tsv("~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WT_Gat201_DEseq2_genes4h.txt")
```

### Filter results 2x different, 5% FDR upregulated
Select rows with estimated log2fold change >0 and -pvalue <0.05
Arrange in descending row estimate, most DE genes at top
New table, similar info to V, just ordered differently, entire table with all stats
Took list of genes (de_WTup_4h$gene)write to new file. Can use this file  GO analysis

```{r}
de_WTup_4h <- dplyr::filter(tresults_4h, estimate > 0, p.adjusted < 0.05) %>%
              dplyr::arrange(desc(estimate))
de_WTup_4h

readr::write_tsv(de_WTup_4h, "~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WTup_DEseq2_padj0p05.txt")
readr::write_lines(de_WTup_4h$gene, "~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WTup_DEseq2_padj0p05_genesonly.txt")
                
```
### WT (GAT201 = yes) vs Gat201 deletion (GAT201 = no) at DOWN 4h only
values negative since down regulated

```{r}
de_WTdown_4h <- dplyr::filter(tresults_4h, estimate < 0, p.adjusted < 0.05) %>%
                dplyr::arrange(estimate)
de_WTdown_4h

readr::write_tsv(de_WTdown_4h, "~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WTdown_DEseq2_padj0p05.txt")
readr::write_lines(de_WTdown_4h$gene, "~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WTdown_DEseq2_padj0p05_genesonly.txt")
```

### Plot log2fold-change (estimate) against baseMean 4h

```{r}
plot_4h<- qplot(x = baseMean, y = estimate, data = tresults_4h, geom = "point")+
               geom_text(aes(label=ifelse(baseMean> 20000, (gene),'')),hjust=0,vjust=0, cex = 1.5, jitter = 0.5)+
               labs(x ="Mean", y= "log2fold change", title = "WT v Gat201 mutants at 4 hrs")
plot_4h
        
```

### Plot fold-change and p-value (volcano) at 4h
this gives the axis only
this lots all genes
this plots up genes only inn colour, others in grey
this plots down genes (will show up and down in diff colours)
change y scale display to show certain p values only to see more
out of bounds squish so can still see, will be squished at the top
plot between 2e-7 and 2e7

```{r volcano 4h}
volcano_4h <- 
  ggplot(data = tresults_4h, aes(x=estimate,y=-log10(p.value))) +
         geom_point(colour = "grey50",size = 0.5) +
         geom_point(data = de_WTup_4h, colour = WTup_colour) +
         geom_point(data = de_WTdown_4h, colour = WTdown_colour) +
         geom_text(aes(label=ifelse(estimate < -2 & -log10(p.value)> 40, (gene),'')),hjust=0,vjust=0, cex = 1.5, jitter = 0.5)+
         geom_text(aes(label=ifelse(estimate > 2 , (gene),'')),hjust=0,vjust=0, cex = 1.5, jitter = 0.5)+
         scale_y_continuous(breaks = c(0,10,20,30,40,50,60,70,80,90,100),
                            limits = c(0,100), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        scale_x_continuous(breaks = seq(-6,6,2),
                            limits = c(-7,7), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        labs(x = "log2-fold change",
        title = "WT vs Gat201 mutants at 4h")
volcano_4h

```

```{r}
ggsave("~/Documents/GitHub/CryptoGat201RNASeq_draft/results_deseq_2/WTvGat201_4h.png",
       width = 5,height=5)
```





