---
title: "Fig 3b Heatmap"
author: "Liz Hughes"
date: "21/12/2021"
output:
  html_document:
    toc: yes
    toc_depth: 2
  word_document:
    toc: yes
    toc_depth: '2'
  pdf_document:
    toc: yes
    toc_depth: '2'
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE , warning=FALSE, message=FALSE)
```

# Summary

This document is an analysis of the Gat201 RNA-seq dataset, comparing the log2foldchange (LFC) between all samples, all timepoints and all media.

Averaging all WT (KN99a and KN99-alpha) strains together and all delta GAT20 (Bahn and Madhani) together.


# Load packages used for analysis

```{r load_packages, include=FALSE}

library(tidyr)
library(readr)
library(dplyr)
library(readxl)
library(stringr)
library(ggplot2)
library(cowplot)
library(biobroom)
library(DESeq2)
library(pheatmap)
library(RColorBrewer)
library(magrittr)
library(markdown)
library(forcats)
library(genefilter)
library(ggrepel)


theme_set(
  theme_cowplot(font_size = 12) +
    theme(panel.grid.major = element_line(colour = "grey80", size = 0.5))
)

```


# Load count data and remove unwanted parts of column names

```{r load_count_data}
# note: we should insist that counts are loaded as integers
# and check the other data types
counts <- readr::read_tsv("~/GitHub/CryptoGat201RNASeq_draft_b/quantseqfwd_EH_050221/counts.txt",
                   comment = "#") %>% 
          dplyr::rename_with(str_remove_all, pattern = "_S[0-9]+_R1_001_aln.bam")

```


# Load sample sheet and format for DESeq2's requirements: readxl::read_excel

Combine columns Strain, Condition and Time to create a new column SCT.
Set the display order for this column (SCT_levels)

```{r load_sample_sheet}

SCT_levels <- c("WT_Y_0", "Mut_Y_0",
                "WT_R_30", "Mut_R_30", "WT_R_120", "Mut_R_120", "WT_R_240", "Mut_R_240",
                "WT_RS_30", "Mut_RS_30","WT_RS_120", "Mut_RS_120","WT_RS_240",   "Mut_RS_240") 


samplesheet <- readxl::read_excel("~/GitHub/CryptoGat201RNASeq_draft_b/input_experiment/Gat201_samplesheet.xlsx") %>%
    magrittr::set_rownames(.$SampleID) %>%
    dplyr::mutate(Strain = 
                    forcats::fct_collapse(Strain, 
                                          "WT" = c("a","A"), 
                                          "Mut" = c("B","M")) %>% 
                    forcats::fct_relevel("del"),
                  Timepoint = factor(Timepoint),
                  SCT = factor(
                    paste( Strain, Condition, Time, sep = "_"),
                    levels = SCT_levels
                    
                  )
    )

```

# Prepare counts_long

Make a tidy long-format count table
Remove unwanted meta data columns
Calculate the log2foldchange of FoldChange in counts_long
Rename FoldChange to LFC in counts_LFC

```{r counts_long}
  counts_long <- counts %>%
                select(-Chr,-Start,-End,-Strand,-Length) %>%
                pivot_longer(!Geneid, 
                              names_to = c("SampleID","SampleNo"), names_sep="_",
                              values_to = "Count") %>%
                group_by(SampleID) %>%
                mutate(TPM = Count / sum(Count) * 1e6) %>%
                ungroup() %>%
                group_by(Geneid) %>%
                mutate(FoldChange = log2(TPM / mean(TPM))) %>%
                ungroup() %>%
                left_join(samplesheet)

  counts_LFC <- dplyr::rename(counts_long, LFC = FoldChange)
  head(counts_LFC)
 

```

# Order counts_LFC by LFC

```{r order_counts_LFC}

counts_LFC_O <- dplyr::arrange(counts_LFC , desc(LFC))
counts_LFC_O
```

# Heatmap using counts_LFC

```{r plot_heatmap_counts_LFC }

   ggplot( data = counts_LFC)  +
           geom_tile(aes(x = SCT, y = Geneid, fill = LFC)) +
            scale_fill_fermenter("log2 fold-change\nWT vs ∆GAT201",
            limits = c(-3.2, 3.2), oob = scales::squish)+
      theme(panel.grid.major = element_blank(),
            axis.text.y = element_blank(),
            axis.line = element_blank(),
            axis.ticks.y = element_blank(),
            axis.text.x = element_text(angle = 90))+
   labs(x = "Condition", y = "Gene") 

```

# Heatmap using counts_LFC_O: ordered by LFC

```{r plot_heatmap_counts_LFC_O}

ggplot( data = counts_LFC_O)  +
           geom_tile(aes(x = SCT, y = Geneid, fill = LFC)) +
            scale_fill_fermenter("log2 fold-change\nWT vs ∆GAT201",
            limits = c(-3.2, 3.2), oob = scales::squish)+
      theme(panel.grid.major = element_blank(),
            axis.text.y = element_blank(),
            axis.line = element_blank(),
            axis.ticks.y = element_blank(),
            axis.text.x = element_text(angle = 90))+
   labs(x = "Condition", y = "Gene")
```


```{r}

```


```{r}

```


```{r}

```


```{r}

```


```{r}

```


```{r}

```


```{r}

```


```{r}

```















