---
title: "PCA analysis of Gat201 RNA-seq dataset"
author: "Liz Hughes and Edward Wallace"
date: "16/12/2021"
output:
  html_document:
    toc: yes
    toc_depth: 2
  word_document:
    toc: yes
    toc_depth: '2'
  pdf_document:
    toc: yes
    toc_depth: '2'
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE , warning=FALSE, message=FALSE)
```

# Summary

This document is a big-picture analysis of trends in the Gat201 RNA-seq dataset.

The first goal is a principal component analysis (PCA) to show the main axes of variation in the data.

# Load packages used for analysis

```{r load_packages, include=FALSE}

library(tidyr)
library(readr)
library(dplyr)
library(readxl)
library(stringr)
library(ggplot2)
library(cowplot)
library(biobroom)
library(DESeq2)
library(pheatmap)
library(RColorBrewer)
library(magrittr)
library(markdown)
library(forcats)
library(genefilter)
library(ggrepel)


theme_set(
  theme_cowplot(font_size = 12) +
    theme(panel.grid.major = element_line(colour = "grey80", size = 0.5))
)

```


# Load count data and remove unwanted parts of column names

```{r load_count_data}
# note: we should insist that counts are loaded as integers
# and check the other data types
counts <- readr::read_tsv("../quantseqfwd_EH_050221/counts.txt",
                   comment = "#") %>% 
          dplyr::rename_with(str_remove_all, pattern = "_S[0-9]+_R1_001_aln.bam")

```


# Load sample sheet and format for DESeq2's requirements: readxl::read_excel

```{r load_sample_sheet}

samplesheet <- readxl::read_excel("../input_experiment/Gat201_samplesheet.xlsx") %>%
    magrittr::set_rownames(.$SampleID) %>%
    dplyr::mutate(GAT201 = 
                    forcats::fct_collapse(Strain, 
                                          "WT" = c("a","A"), 
                                          "∆" = c("B","M")) %>% 
                    forcats::fct_relevel("∆")
                  ) 

```

# Select counts

```{r select_counts_all}

counts_all <-
      dplyr::select(counts, samplesheet$SampleID) %>%
      magrittr::set_rownames(counts$Geneid)

```

# Construct the data object from the matrix of counts and the samplesheet information table

```{r dds_counts_all}

dds_all <- DESeqDataSetFromMatrix(countData = counts_all,
                                   colData = samplesheet,
                                   design = ~ Condition  + Time + GAT201)

```


# rlog transformation

```{r rlog_dds_all}

rld_all <- rlog(dds_all)
head(assay(rld_all))

```


## Calculate principal components of rlog

```{r calculate_pca_rlog}

## calculate the principal components sample-wise
pca_rlog <- prcomp(t(assay(rld_all)))

# combine pc projections to samples with sample sheet
pcdf_rlog <- bind_cols(
  as_tibble(colData(rld_all)) %>%
    mutate(GAT201_Cond = interaction(GAT201, Condition, sep = "_"),
           GAT201_Serum = forcats::fct_collapse(
             GAT201_Cond,
             "∆_none"   = c("∆_R", "∆_Y"),
             "WT_none"   = c("WT_R", "WT_Y"),
             "∆_Serum" = "∆_RS",
             "WT_Serum"  = "WT_RS"
           ),
    ),
  as_tibble(pca_rlog$x)
)

propvar_rlog_df <- tibble(
  PC = seq.int(1L, ncol(pca_rlog$x) ),
  prop_var = pca_rlog$sdev^2/sum(pca_rlog$sdev^2)
)
```


# Plot PCA percentage variance explained to find how many PCs to pay attention to.

```{r plot_percentvar, fig.width = 3, fig.height = 2}
ggplot(data = propvar_rlog_df, 
       aes(x = PC, y = prop_var)) +
  geom_col(fill = "blue") +
  scale_x_continuous("principal component",
                     limits = c(0.4,10.6), 
                     # hack xlim to include the bars and a little space.
                     breaks = 1L:10L,
                     expand = c(0,0)) + 
  scale_y_continuous("prop. of variance", expand = c(0,0)) +
  theme(panel.grid.major = element_blank())
```



# Plot PC1 vs PC2 with default values

```{r plot_PC12_default}
ggplot(data = pcdf_rlog,
       aes(colour = factor(Time), 
           shape = GAT201_Cond)
       ) +
  geom_point(aes(x = PC1, y = PC2)) 
```

# Plot PC1 vs PC2 prioritising Time and GAT201_serum

```{r plot_PC12_fancy}
plot_PCA_Time_GC <- function(xpc = "PC1", ypc = "PC2",
                             data = pcdf_rlog) {
  # Plots PCA with default settings to highlight:
  #  Time as colour
  #  GAT201_Serum as shape, i.e. interaction of GAT201 and serum 
  #    (treating R and Y both as "no serum")
  # 
  # Default data input is pcdf_rlog, but it should work with others
  # as long as they have Time and GAT201_Serum as variables.
  
  ggplot(data = data,
         aes(colour = factor(Time), 
             shape = GAT201_Serum)
  ) +
    geom_vline(xintercept = 0) +
    geom_hline(yintercept = 0) +
    geom_point(aes_string(x = xpc, y = ypc)) +
    scale_shape_manual("Gat201, Serum",
                       values = c("∆_none"   = 0,
                                  "WT_none"   = 15,
                                  "∆_Serum" = 2,
                                  "WT_Serum"  = 17),
                       labels = c("∆_none"   = "∆,  none",
                                  "WT_none"   = "WT, none",
                                  "∆_Serum" = "∆,  + serum",
                                  "WT_Serum"  = "WT, + serum")
    ) +
    scale_colour_manual("Time (mins.)",
                        values = c("0" = "grey20",
                                   "30" = "darkblue",
                                   "120" = "purple2",
                                   "240" = "darkred")) +
    theme(axis.line = element_blank(), 
          panel.grid.major = element_blank(),
          panel.border = element_rect(colour= "grey50", size = 0.5))
  
}
plot_PCA_Time_GC()
```


## To do

- make nice aligned plots of first few PCs
- try again with vst and see if it's different in any important way
