---
title: "PCA Plots"
author: "Liz Hughes"
date: "23/11/2021"
output:
  html_document:
    toc: yes
    toc_depth: 2
  word_document:
    toc: yes
    toc_depth: '2'
  pdf_document:
    toc: yes
    toc_depth: '2'
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE , warning=FALSE, message=FALSE)
```

# Load packages used for analysis
```{r Load packages used for analysis, include=FALSE}

library(tidyr)
library(readr)
library(dplyr)
library(readxl)
library(stringr)
library(ggplot2)
library(cowplot)
library(biobroom)
library(DESeq2)
library(pheatmap)
library(RColorBrewer)
library(magrittr)
library(markdown)
library(forcats)
library(genefilter)
library(ggrepel)


theme_set(
  theme_cowplot(font_size = 12) +
    theme(panel.grid.major = element_line(colour = "grey80", size = 0.5))
)

```


# Load count data and remove unwanted parts of column names

```{r Load count data}
counts <- readr::read_tsv("~/GitHub/CryptoGat201RNASeq_draft_b/quantseqfwd_EH_050221/counts.txt",
                   comment = "#") %>% 
          dplyr::rename_with(str_remove_all, pattern = "_S[0-9]+_R1_001_aln.bam")

```


# Load sample sheet and format for DESeq2's requirements: readxl::read_excel

```{r Load sample sheet}
samplesheet <- readxl::read_excel("~/GitHub/CryptoGat201RNASeq_draft_b/input_experiment/Gat201_samplesheet.xlsx") %>%
    magrittr::set_rownames(.$SampleID)%>%
    dplyr::mutate(GAT201 = forcats::fct_collapse(Strain, WT = c("a","A"), Gat201 = c("B","M")))

```


# Select counts

```{r Select counts_all}
counts_all <-
  dplyr::select(counts, samplesheet$SampleID) %>%
   magrittr::set_rownames(counts$Geneid)

```

# Construct the data object from the matrix of counts and the samplesheet information table

```{r dds_all_counts_all}

(dds_all <- DESeqDataSetFromMatrix(countData = counts_all,
                                  colData = samplesheet,
                                  design = ~ Condition  + Time + GAT201))

```

# rlog Transformation

```{r rlog_dds_all}

rld_all <- rlog(dds_all)
           head(assay(rld_all))
           
```





# PCA Plot for All, 0 Hours, 30 Mins, 2 Hours and 4 Hours using rlog transformation

## Plot PCA for All

```{r PC1-5}
ntop <- 500
rv <- rowVars(assay(rld_all))
select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, length(rv)))]
mat <- t( assay(rld_all)[select, ] )

pca <-prcomp(mat)
pca <- as.data.frame(pca$x)

pca$PC1
pca$PC2
pca$PC3
pca$PC4
pca$PC5

percentVar <- pca$sdev^2 / sum( pca$sdev^2 )

```


```{r ggplot}

 ggplot(data = pca,
              aes(x = PC1, y = PC2)) +
    geom_point() +
    
    labs(x="PC1", 
         y="PC2", 
         title = "PCA Analysis") +
    theme(text = element_text(size = 22))

 ggplot(data = pca,
              aes(x = PC2, y = PC3)) +
    geom_point() +
    
    labs(x="PC2", 
         y="PC3", 
         title = "PCA Analysis") +
    theme(text = element_text(size = 22))

```

```{r function}

#getMethod("plotPCA","DESeqTransform")
#DESeq2:::plotPCA.DESeqTransform

plotPCA <- function (object, intgroup = "condition", ntop = 500, returnData = FALSE) 
{
    rv <- rowVars(assay(object))
    select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, 
        length(rv)))]
    pca <- prcomp(t(assay(object)[select, ]))
    percentVar <- pca$sdev^2/sum(pca$sdev^2)
    if (!all(intgroup %in% names(colData(object)))) {
        stop("the argument 'intgroup' should specify columns of colData(dds)")
    }
    intgroup.df <- as.data.frame(colData(object)[, intgroup, 
        drop = FALSE])
    group <- if (length(intgroup) > 1) {
        factor(apply(intgroup.df, 1, paste, collapse = " : "))
    }
    else {
        colData(object)[[intgroup]]
    }
    d <- data.frame(PC1 = pca$x[, 1], PC2 = pca$x[, 2], PC3 = pca$x[, 3], PC4 = pca$x[, 4], PC5 = pca$x[, 5], group = group, 
        intgroup.df, name = colnames(object))
    if (returnData) {
        attr(d, "percentVar") <- percentVar[4:5]
        return(d)
    }
    ggplot(data = d, aes_string(x = "PC4", y = "PC5", color = "group")) + 
        geom_point(size = 3) + xlab(paste0("PC4: ", round(percentVar[4] * 
        100), "% variance")) + ylab(paste0("PC5: ", round(percentVar[5] * 
        100), "% variance")) + coord_fixed()
    
  }


```





```{r PC23}
PC23_Time <- print(plotPCA(rld_all, intgroup = c("GAT201", "Time")))
PC23_Cond <- print(plotPCA(rld_all, intgroup = c("GAT201", "Condition")))

```


```{r PC34}
PC34_Time <- print(plotPCA(rld_all, intgroup = c("GAT201", "Time")))
PC34_Cond <-print(plotPCA(rld_all, intgroup = c("GAT201", "Condition")))

```


```{r PC45}
PC45_Time <-print(plotPCA(rld_all, intgroup = c("GAT201", "Time")))
PC45_Cond <-print(plotPCA(rld_all, intgroup = c("GAT201", "Condition")))

```






```{r PCA_Plot_All}



plotPCA(rld_all, intgroup = c("Strain", "Condition", "Time"))

plotPCA(rld_all, intgroup = c("GAT201", "Condition", "Time"))

plotPCA(rld_all, intgroup = c("GAT201", "Condition"))

plotPCA(rld_all, intgroup = c("GAT201", "Time"))

plotPCA(rld_all, intgroup = c("Time", "Condition"))

plotPCA(rld_all, intgroup = c("Strain"))

plotPCA(rld_all, intgroup = c("GAT201"))

plotPCA(rld_all, intgroup = c("Condition"))

plotPCA(rld_all, intgroup = c("Time"))
       
```


## Plot PCA for 0h

```{r PCA_Plot_rld_0h}
samplesheet_0h <-  dplyr::filter(samplesheet, Time == "0")
                  print(samplesheet_0h)

counts_0h <-  dplyr::select(counts_all, samplesheet_0h$SampleID) %>%
              magrittr::set_rownames(counts$Geneid)

(dds0h <- DESeqDataSetFromMatrix(countData = counts_0h,
                                 colData = samplesheet_0h,
                                 design = ~ Strain))

rld_0h <- rlog(dds0h)
          head(assay(rld_0h))

plotPCA(rld_0h, intgroup = c("Strain", "Condition"))

       
```

## Plot PCA for 30min
```{r PCA Plot rld_30m}
samplesheet_30m <- dplyr::filter(samplesheet, Time == "30")
                   print(samplesheet_30m)

counts_30m <- dplyr::select(counts, samplesheet_30m$SampleID) %>%
              magrittr::set_rownames(counts$Geneid)

(dds30m <- DESeqDataSetFromMatrix(countData = counts_30m,
                                 colData = samplesheet_30m,
                                 design = ~ Condition + Strain))

rld_30m <- rlog(dds30m)
           head(assay(rld_30m))

plotPCA(rld_30m, intgroup = c("Strain", "Condition"))

```

## Plot PCA for 2h
```{r PCA_Plot_rld_2h}
samplesheet_2h <- dplyr::filter(samplesheet, Time == "120")
                  print(samplesheet_2h)

counts_2h <-   dplyr::select(counts, samplesheet_2h$SampleID) %>%
               magrittr::set_rownames(counts$Geneid)

(dds2h <- DESeqDataSetFromMatrix(countData = counts_2h,
                                 colData = samplesheet_2h,
                                 design = ~ Condition + Strain))

rld_2h <- rlog(dds2h)
          head(assay(rld_2h))

plotPCA(rld_2h, intgroup = c("Strain", "Condition"))

```

## Plot PCA for 4h
```{r PCA_Plot_rld_4h}
samplesheet_4h <- dplyr::filter(samplesheet, Time == "240")
                  print(samplesheet_4h)

counts_4h <-   dplyr::select(counts, samplesheet_4h$SampleID) %>%
               magrittr::set_rownames(counts$Geneid)

(dds4h <- DESeqDataSetFromMatrix(countData = counts_4h,
                                 colData = samplesheet_4h,
                                 design = ~ Condition + Strain))

rld_4h <- rlog(dds4h)
          head(assay(rld_4h))

plotPCA(rld_4h, intgroup = c("Strain", "Condition"))

```


# PCA Plot For genes with min. TPM>20 at 0 Hours, 30 Mins, 2 Hours and 4 Hours

## Make a tidy long-format count table, including normalized TPM values and Fold Change

```{r Make a tidy long-format count table}

counts_long <- counts %>%
              set_names(str_remove(names(counts), "_R1_001_aln.bam")) %>%
              select(-Chr,-Start,-End,-Strand,-Length) %>%
              pivot_longer(!Geneid, 
                            names_to = c("SampleID","SampleNo"), names_sep="_",
                            values_to = "Count") %>%
              group_by(SampleID) %>%
              mutate(TPM = Count / sum(Count) * 1e6) %>%
              ungroup() %>%
              group_by(Geneid) %>%
              mutate(FoldChange = TPM / mean(TPM)) %>%
              ungroup() %>%
              left_join(samplesheet)
head(counts_long)
```

## Write long-format counts to an output file

```{r Write long-format counts to an output file}

counts_long  %>%
    select(Geneid,SampleID,Strain,Condition,Time,BioRep,Count,TPM,FoldChange) %>%
    mutate(TPM = round(TPM, digits = 3)) %>%
    dplyr::mutate(GAT201 = forcats::fct_collapse(Strain, Yes = c("a","A"), No = c("B","M")))

```


## Reshape the TPMs into a "wide" table that we can use for heatmap

```{r Reshape the TPMs into a wide table}

tpms_wide <- counts_long %>%
              pivot_wider(id_cols = Geneid, names_from = SampleID, values_from = TPM)


```

## Calculate a summary table of mean, min, max TPM by gene

```{r TPM summary table}
genesummary <- 
    counts_long %>%
    group_by(Geneid) %>%
    summarise(TPM.mean = mean(TPM), TPM.min = min(TPM), TPM.max = max(TPM))
genesummary
```


## Create lists with TPM values all above thresholds 

```{r TPM thresholds}
genelist_mintpm20 <- filter(genesummary,TPM.min > 20)$Geneid
genelist_mintpm100 <- filter(genesummary,TPM.min > 100)$Geneid
genelist_mintpm500 <- filter(genesummary,TPM.min > 500)$Geneid
genelist_mintpm500

```


## Reshape the fold changes into a "wide" table to use for PCA

```{r Reshape the fold changes into a "wide" table}

foldchange_wide <- counts_long %>%
                   pivot_wider(id_cols = Geneid, names_from = SampleID, values_from = FoldChange)

```


## Principal Component Analysis 

## For genes with min. TPM 20

```{r PCA TPM 20}
pca_log10foldchange_mintpm20 <- 
    foldchange_wide %>%
    filter(Geneid %in% genelist_mintpm20) %>%
    .[,-1] %>%
    log10() %>%
    prcomp(scale=TRUE)
pca_log10foldchange_mintpm20
  
```

## Plot first and second principal components

```{r PC12a}
plot_pca_PC12_Condition <- 
pca_log10foldchange_mintpm20 %>%
    broom::tidy(matrix = "rotation") 
plot_pca_PC12_Condition
  
plot_pca_PC12_Conditionb <- dplyr::rename(plot_pca_PC12_Condition, SampleID = column) %>%
    left_join(samplesheet) %>%
    pivot_wider(names_from = "PC", names_prefix = "PC", values_from = "value") %>%
  
    ggplot(aes(PC1, PC2))+
    geom_vline(aes(xintercept = 0),size=0.2,colour = "grey50") +
    geom_hline(aes(yintercept = 0),size=0.2,colour = "grey50")  +
    geom_text(
        aes(label = Timepoint, colour=Condition),
        key_glyph = "point",
    ) +
    coord_equal()

plot_pca_PC12_Conditionb

ggsave("~/GitHub/CryptoGat201RNASeq_draft_Liz/results_counts/plot_pca_PC12_Conditionb.jpeg",
       plot_pca_PC12_Conditionb,width = 4,height=4)

```

## Plot second and third principal components 

```{r PC23a}
plot_pca_PC23_Condition <- 
pca_log10foldchange_mintpm20 %>%
    broom::tidy(matrix = "rotation") 
plot_pca_PC23_Condition
  
plot_pca_PC23_Conditionb <- dplyr::rename(plot_pca_PC23_Condition, SampleID = column) %>%
    left_join(samplesheet) %>%
    pivot_wider(names_from = "PC", names_prefix = "PC", values_from = "value") %>%
  
    ggplot(aes(PC2, PC3))+
    geom_vline(aes(xintercept = 0),size=0.2,colour = "grey50") +
    geom_hline(aes(yintercept = 0),size=0.2,colour = "grey50")  +
    geom_text(
        aes(label = Timepoint, colour=Condition),
        key_glyph = "point",
    ) +
    coord_equal()

plot_pca_PC23_Conditionb


```

## Plot third and fourth principal components 

```{r PC34a}
plot_pca_PC34_Condition <- 
pca_log10foldchange_mintpm20 %>%
    broom::tidy(matrix = "rotation") 
plot_pca_PC34_Condition
  
plot_pca_PC34_Conditionb <- dplyr::rename(plot_pca_PC34_Condition, SampleID = column) %>%
    left_join(samplesheet) %>%
    pivot_wider(names_from = "PC", names_prefix = "PC", values_from = "value") %>%
  
    ggplot(aes(PC3, PC4))+
    geom_vline(aes(xintercept = 0),size=0.2,colour = "grey50") +
    geom_hline(aes(yintercept = 0),size=0.2,colour = "grey50")  +
    geom_text(
        aes(label = Timepoint, colour=Condition),
        key_glyph = "point",
    ) +
    coord_equal()

plot_pca_PC34_Conditionb

```

## Plot  fourth and fifthprincipal components

```{r PC45a}
plot_pca_PC45_Condition <- 
pca_log10foldchange_mintpm20 %>%
    broom::tidy(matrix = "rotation") 
plot_pca_PC45_Condition
  
plot_pca_PC45_Conditionb <- dplyr::rename(plot_pca_PC45_Condition, SampleID = column) %>%
    left_join(samplesheet) %>%
    pivot_wider(names_from = "PC", names_prefix = "PC", values_from = "value") %>%
  
    ggplot(aes(PC4, PC5))+
    geom_vline(aes(xintercept = 0),size=0.2,colour = "grey50") +
    geom_hline(aes(yintercept = 0),size=0.2,colour = "grey50")  +
    geom_text(
        aes(label = Timepoint, colour=Condition),
        key_glyph = "point",
    ) +
    coord_equal()

plot_pca_PC45_Conditionb

```

## Plot PCA loadings, i.e. percent variance explained by each PC

```{r Plot PCA loadings}
plot_pcaloading_log10foldchange_mintpm20 <- 
pca_log10foldchange_mintpm20 %>%
    broom::tidy(matrix = "d") %>%
    ggplot(aes(PC, percent)) +
    geom_col(fill="blue")   + 
    scale_x_continuous(breaks = 1:12,limits = c(0.5,12.5)) +
    scale_y_continuous(
        labels = scales::percent_format(),
        expand = c(0, 0)
    ) +
    theme_minimal_hgrid(12)

plot_pcaloading_log10foldchange_mintpm20

```
