---
title: "Volcano plots and DEG"
author: "Liz Hughes"
date: "24/11/2021"
output:
  html_document:
    toc: yes
    toc_depth: 2
  word_document:
    toc: yes
    toc_depth: '2'
  pdf_document:
    toc: yes
    toc_depth: '2'
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE , warning=FALSE, message=FALSE)
```

# Load packages used for analysis
```{r Load packages used for analysis, include=FALSE}

library(tidyr)
library(readr)
library(dplyr)
library(readxl)
library(stringr)
library(ggplot2)
library(cowplot)
library(biobroom)
library(DESeq2)
library(pheatmap)
library(RColorBrewer)
library(magrittr)
library(markdown)
library(forcats)
library(ggrepel)


theme_set(
  theme_cowplot(font_size = 12) +
    theme(panel.grid.major = element_line(colour = "grey80", size = 0.5))
)

```


# Load count data and remove unwanted parts of column names

```{r Load count data}

counts <- readr::read_tsv("~/GitHub/CryptoGat201RNASeq_draft_b/quantseqfwd_EH_050221/counts.txt",
                   comment = "#") %>% 
          dplyr::rename_with(str_remove_all, pattern = "_S[0-9]+_R1_001_aln.bam")

```

# Load sample sheet and format for DESeq2's requirements: readxl::read_excel

```{r Load sample sheet}

samplesheet <- readxl::read_excel("~/GitHub/CryptoGat201RNASeq_draft_b/input_experiment/Gat201_samplesheet.xlsx") %>%
    magrittr::set_rownames(.$SampleID)%>%
    dplyr::mutate(GAT201 = forcats::fct_collapse(Strain, Yes = c("a","A"), No = c("B","M")))
```


# Define colours for specific samples

```{r define colours}
WTup_colour   = "orange"
WTdown_colour = "darkgreen"
Gat201up_colour  = "green"
Gat201down_colour = "purple"
```


# Volcano plots and up/down DEG

## Gat201 deletionv WT 0h DEG
```{r DESeq0h}

samplesheet_0h <- dplyr::filter(samplesheet, Time == "0")
                  print(samplesheet_0h)

counts_0h <-   dplyr::select(counts, samplesheet_0h$SampleID) %>%
               magrittr::set_rownames(counts$Geneid)

dds_0h <- DESeqDataSetFromMatrix(countData = counts_0h,
                                 colData = samplesheet_0h,
                                 design = ~ GAT201)%>%
        DESeq()


resultsNames(dds_0h)

results_0h <- results(dds_0h, name="GAT201_No_vs_Yes")

tresults_0h <- biobroom::tidy.DESeqResults(results_0h)
head(tresults_0h)

tresults_0h %>%  
dplyr::mutate(minus_log10_pvalue = -log10(p.value)) %>%
readr::write_tsv("~/GitHub/CryptoGat201RNASeq_draft_b/results_deseq/Gat201_WT_0h.txt")

de_Gat201up_0h <- dplyr::filter(tresults_0h, estimate > 1, p.adjusted < 0.05) %>%
              dplyr::arrange(desc(estimate))
de_Gat201up_0h

readr::write_tsv(de_Gat201up_0h, "~/GitHub/CryptoGat201RNASeq_draft_b/results_deseq/Gat201up_padj05_0h.txt")

readr::write_lines(de_Gat201up_0h$gene, "~/GitHub/CryptoGat201RNASeq_draft_b/results_deseq/Gat201up_padj05_0h_genesonly.csv")


de_Gat201down_0h <- dplyr::filter(tresults_0h, estimate < -1, p.adjusted < 0.05) %>%
                dplyr::arrange(estimate)
de_Gat201down_0h

readr::write_tsv(de_Gat201down_0h, "~/GitHub/CryptoGat201RNASeq_draft_b/results_deseq/Gat201down_padj05_0h.txt")

readr::write_lines(de_Gat201down_0h$gene, "~/GitHub/CryptoGat201RNASeq_draft_B/results_deseq/Gat201down_padj05_0h_genesonly.csv")
```

## Gat201 deletionv WT 0h Volcano Plot

```{r volcano_0h}

#Volcano plot with significant genes labelled

volcano_0h <- 
  ggplot(data = tresults_0h, aes(x=estimate,y=-log10(p.value))) +
         geom_point(colour = "grey50",size = 0.5) +
         geom_point(data = de_Gat201up_0h, colour = Gat201up_colour) +
         geom_point(data = de_Gat201down_0h, colour = Gat201down_colour) +
         
         geom_text_repel(aes(label=ifelse(estimate < -1 & -log10(p.value)> 1.3, (gene),'')), size= 2, max.overlaps = getOption("ggrepel.max.overlaps", default = 15))+
         geom_text_repel(aes(label=ifelse(estimate > 1 & -log10(p.value)> 1.3, (gene),'')), size = 2, max.overlaps = getOption("ggrepel.max.overlaps", default = 15))+
  
         scale_y_continuous(breaks = c(0,5,10,15,20),
                            limits = c(0,20), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        scale_x_continuous(breaks = seq(-5,5,1),
                            limits = c(-5,5), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        labs(x = "log2-fold change",
             y = "-log10(p.value)",
        title = "Gat201 mutants v WT at 0h")
volcano_0h

#volcano plot with Gat201 and possible downstream regulated genes

volcano_0hsp <- 
  ggplot(data = tresults_0h, aes(x=estimate,y=-log10(p.value))) +
         geom_point(colour = "grey50",size = 0.5) +
         geom_point(data = de_Gat201up_0h, colour = Gat201up_colour) +
         geom_point(data = de_Gat201down_0h, colour = Gat201down_colour) +
    geom_text_repel(data = filter(tresults_0h, gene =="CNAG_01551"), label="Gat201",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_0h, gene =="CNAG_06762"), label="Gat204",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_0h, gene =="CNAG_063641"), label="Blp1",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_0h, gene =="CNAG_00883"), label="Ecm2201",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_0h, gene =="CNAG_05835"), label="Liv3",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_0h, gene =="CNAG_03012"), label="CQS1",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_0h, gene =="CNAG_01681"), label="FCY2",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_0h, gene =="CNAG_06401"), label="Ric8",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_0h, gene =="CNAG_07470"), label="Pde2",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_0h, gene =="CNAG_06241"), label="Cfo1",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_0h, gene =="CNAG_02777"), label="Pho84",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_0h, gene =="CNAG_04755"), label="Bck1",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_0h, gene =="CNAG_02357"), label="Mkk2",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_0h, gene =="CNAG_04514"), label="Mpk1",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_0h, gene =="CNAG_04969"), label="UGD1",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_0h, gene =="CNAG_01626"), label="Ada2",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_0h, gene =="CNAG_03024"), label="Rim15",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_0h, gene =="CNAG_04864"), label="Cir1",hjust = 1.1,vjust = -0.1, colour= "red")+
         scale_y_continuous(breaks = c(0,5,10,15,20),
                            limits = c(0,20), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        scale_x_continuous(breaks = seq(-5,5,1),
                            limits = c(-5,5), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        labs(x = "log2-fold change",
             y = "-log10(p.value)",
             title = "Gat201 mutants v WT at 0h")
volcano_0hsp


```

## Gat201 deletionv WT 30m DEG

```{r DESeq30m}

samplesheet_30m <- dplyr::filter(samplesheet, Time == "30")
                  print(samplesheet_30m)

counts_30m <-   dplyr::select(counts, samplesheet_30m$SampleID) %>%
               magrittr::set_rownames(counts$Geneid)

dds_30m <- DESeqDataSetFromMatrix(countData = counts_30m,
                                 colData = samplesheet_30m,
                                 design = ~ GAT201)%>%
        DESeq()

resultsNames(dds_30m)
results_30m <- results(dds_30m, name="GAT201_No_vs_Yes")

tresults_30m <- biobroom::tidy.DESeqResults(results_30m)
head(tresults_30m)

tresults_30m %>%  
dplyr::mutate(minus_log10_pvalue = -log10(p.value)) %>%
readr::write_tsv("~/GitHub/CryptoGat201RNASeq_draft_b/results_deseq/Gat201_WT_30m.txt")

de_Gat201up_30m <- dplyr::filter(tresults_30m, estimate > 1, p.adjusted < 0.05) %>%
              dplyr::arrange(desc(estimate))
de_Gat201up_30m

readr::write_tsv(de_Gat201up_30m, "~/GitHub/CryptoGat201RNASeq_draft_b/results_deseq/Gat201up_padj05_30m.txt")

readr::write_lines(de_Gat201up_30m$gene, "~/GitHub/CryptoGat201RNASeq_draft_b/results_deseq/Gat201up_padj05_30m_genesonly.csv")

de_Gat201down_30m <- dplyr::filter(tresults_30m, estimate < -1, p.adjusted < 0.05) %>%
                dplyr::arrange(estimate)
de_Gat201down_30m

readr::write_tsv(de_Gat201down_30m, "~/GitHub/CryptoGat201RNASeq_draft_b/results_deseq/Gat201down_padj05_30m.txt")

readr::write_lines(de_Gat201down_30m$gene, "~/GitHub/CryptoGat201RNASeq_draft_b/results_deseq/Gat201down_padj05_30m_genesonly.csv")

```


## Gat201 deletionv WT 30m Volcano Plot

```{r volcano_30m}

#Volcano plot with significant genes labelled

volcano_30m <- 
  ggplot(data = tresults_30m, aes(x=estimate,y=-log10(p.value))) +
         geom_point(colour = "grey50",size = 0.5) +
         geom_point(data = de_Gat201up_30m, colour = Gat201up_colour) +
         geom_point(data = de_Gat201down_30m, colour = Gat201down_colour) +
         geom_text_repel(aes(label=ifelse(estimate < -1 & -log10(p.value)> 1.3, (gene),'')), size= 2, max.overlaps = getOption("ggrepel.max.overlaps", default = 15))+
         geom_text_repel(aes(label=ifelse(estimate > 1 & -log10(p.value)> 1.3, (gene),'')), size = 2, max.overlaps = getOption("ggrepel.max.overlaps", default = 15))+
         scale_y_continuous(breaks = c(0,10,20,30,40,50,60),
                            limits = c(0,60), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        scale_x_continuous(breaks = seq(-5,5,1),
                            limits = c(-5,5), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        labs(x = "log2-fold change",
        title = "Gat201 mutants v WT at 30 min")
volcano_30m

#volcano plot with Gat201 and possible downstream regulated genes

volcano_30msp <- 
  ggplot(data = tresults_30m, aes(x=estimate,y=-log10(p.value))) +
         geom_point(colour = "grey50",size = 0.5) +
         geom_point(data = de_Gat201up_30m, colour = Gat201up_colour) +
         geom_point(data = de_Gat201down_30m, colour = Gat201down_colour) +
    geom_text_repel(data = filter(tresults_30m, gene =="CNAG_01551"), label="Gat201",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_30m, gene =="CNAG_06762"), label="Gat204",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_30m, gene =="CNAG_063641"), label="Blp1",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_30m, gene =="CNAG_00883"), label="Ecm2201",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_30m, gene =="CNAG_05835"), label="Liv3",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_30m, gene =="CNAG_03012"), label="CQS1",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_30m, gene =="CNAG_01681"), label="FCY2",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_30m, gene =="CNAG_06401"), label="Ric8",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_30m, gene =="CNAG_07470"), label="Pde2",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_30m, gene =="CNAG_06241"), label="Cfo1",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_30m, gene =="CNAG_02777"), label="Pho84",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_30m, gene =="CNAG_04755"), label="Bck1",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_30m, gene =="CNAG_02357"), label="Mkk2",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_30m, gene =="CNAG_04514"), label="Mpk1",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_30m, gene =="CNAG_04969"), label="UGD1",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_30m, gene =="CNAG_01626"), label="Ada2",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_30m, gene =="CNAG_03024"), label="Rim15",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_30m, gene =="CNAG_04864"), label="Cir1",hjust = 1.1,vjust = -0.1, colour= "red")+
          scale_y_continuous(breaks = c(0,10,20,30,40,50,60),
                            limits = c(0,60), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        scale_x_continuous(breaks = seq(-5,5,1),
                            limits = c(-5,5), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        labs(x = "log2-fold change",
        title = "Gat201 mutants v WT at 30 min")
volcano_30msp

```

## Gat201 deletionv WT 2h DEG

```{r DESeq2h}

samplesheet_2h <- dplyr::filter(samplesheet, Time == "120")
                  print(samplesheet_2h)

counts_2h <-   dplyr::select(counts, samplesheet_2h$SampleID) %>%
               magrittr::set_rownames(counts$Geneid)

dds_2h <- DESeqDataSetFromMatrix(countData = counts_2h,
                                 colData = samplesheet_2h,
                                 design = ~ GAT201)%>%
        DESeq()

resultsNames(dds_2h)
results_2h <- results(dds_2h, name="GAT201_No_vs_Yes")

tresults_2h <- biobroom::tidy.DESeqResults(results_2h)
head(tresults_2h)

ggplot(data=tresults_2h, aes(baseMean,estimate))+
      geom_point()

tresults_2h %>%  
dplyr::mutate(minus_log10_pvalue = -log10(p.value)) %>%
readr::write_tsv("~/GitHub/CryptoGat201RNASeq_draft_b/results_deseq/Gat201_WT_2h.txt")

de_Gat201up_2h <- dplyr::filter(tresults_2h, estimate > 1, p.adjusted < 0.05) %>%
              dplyr::arrange(desc(estimate))
de_Gat201up_2h

readr::write_tsv(de_Gat201up_2h, "~/GitHub/CryptoGat201RNASeq_draft_b/results_deseq/Gat201up_padj05_2h.txt")

readr::write_lines(de_Gat201up_2h$gene, "~/GitHub/CryptoGat201RNASeq_draft_b/results_deseq/Gat201up_padj05_2h_genesonly.csv")

de_Gat201down_2h <- dplyr::filter(tresults_2h, estimate < -1, p.adjusted < 0.05) %>%
                dplyr::arrange(estimate)
de_Gat201down_2h

readr::write_tsv(de_Gat201down_2h, "~/GitHub/CryptoGat201RNASeq_draft_b/results_deseq/Gat201down_padj05_2h.txt")

readr::write_lines(de_Gat201down_2h$gene, "~/GitHub/CryptoGat201RNASeq_draft_b/results_deseq/Gat201down_padj05_2h_genesonly.csv")

```

## Gat201 deletionv WT 2h Volcano Plot

```{r volcano_2h}

#Volcano plot with significant genes labelled

volcano_2h <- 
  ggplot(data = tresults_2h, aes(x=estimate,y=-log10(p.value))) +
         geom_point(colour = "grey50",size = 0.5) +
         geom_point(data = de_Gat201up_2h, colour = Gat201up_colour) +
         geom_point(data = de_Gat201down_2h, colour = Gat201down_colour) +
         geom_text_repel(aes(label=ifelse(estimate < -1 & -log10(p.value)> 1.3, (gene),'')), size= 2, max.overlaps = getOption("ggrepel.max.overlaps", default = 15))+
         geom_text_repel(aes(label=ifelse(estimate > 1 & -log10(p.value)> 1.3, (gene),'')), size = 2, max.overlaps = getOption("ggrepel.max.overlaps", default = 15))+
         scale_y_continuous(breaks = c(0,10,20,30,40,50,60,70,80,90,100),
                            limits = c(0,100), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        scale_x_continuous(breaks = seq(-5,5,1),
                            limits = c(-5,5), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        labs(x = "log2-fold change",
        title = "Gat201 mutants v WT at 2h")
volcano_2h


#volcano plot with Gat201 and possible downstream regulated genes

volcano2hsp <- 
  ggplot(data = tresults_2h, aes(x=estimate,y=-log10(p.value))) +
         geom_point(colour = "grey50",size = 0.5) +
         geom_point(data = de_Gat201up_2h, colour = Gat201up_colour) +
         geom_point(data = de_Gat201down_2h, colour = Gat201down_colour) +
    geom_text_repel(data = filter(tresults_2h, gene =="CNAG_01551"), label="Gat201",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_2h, gene =="CNAG_06762"), label="Gat204",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_2h, gene =="CNAG_063641"), label="Blp1",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_2h, gene =="CNAG_00883"), label="Ecm2201",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_2h, gene =="CNAG_05835"), label="Liv3",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_2h, gene =="CNAG_03012"), label="CQS1",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_2h, gene =="CNAG_01681"), label="FCY2",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_2h, gene =="CNAG_06401"), label="Ric8",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_2h, gene =="CNAG_07470"), label="Pde2",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_2h, gene =="CNAG_06241"), label="Cfo1",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_2h, gene =="CNAG_02777"), label="Pho84",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_2h, gene =="CNAG_04755"), label="Bck1",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_2h, gene =="CNAG_02357"), label="Mkk2",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_2h, gene =="CNAG_04514"), label="Mpk1",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_2h, gene =="CNAG_04969"), label="UGD1",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_2h, gene =="CNAG_01626"), label="Ada2",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_2h, gene =="CNAG_03024"), label="Rim15",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_2h, gene =="CNAG_04864"), label="Cir1",hjust = 1.1,vjust = -0.1, colour= "red")+
         scale_y_continuous(breaks = c(0,10,20,30,40,50,60,70),
                            limits = c(0,70), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        scale_x_continuous(breaks = seq(-5,5,1),
                            limits = c(-5,5), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        labs(x = "log2-fold change",
             y = "-log10(p.value)",
             title = "Gat201 mutants v WT at 2h")
volcano2hsp

```


## Gat201 deletionv WT 4h DEG

```{r DESeq4h}

samplesheet_4h <- dplyr::filter(samplesheet, Time == "240")
                  print(samplesheet_4h)

counts_4h <-   dplyr::select(counts, samplesheet_4h$SampleID) %>%
               magrittr::set_rownames(counts$Geneid)

dds_4h <- DESeqDataSetFromMatrix(countData = counts_4h,
                                 colData = samplesheet_4h,
                                 design = ~ GAT201)%>%
        DESeq()

resultsNames(dds_4h)
results_4h <- results(dds_4h, name="GAT201_No_vs_Yes")

tresults_4h <- biobroom::tidy.DESeqResults(results_4h)
head(tresults_4h)

tresults_4h %>%  
dplyr::mutate(minus_log10_pvalue = -log10(p.value)) %>%
readr::write_tsv("~/GitHub/CryptoGat201RNASeq_draft_b/results_deseq/Gat201_WT_4h.txt")

de_Gat201up_4h <- dplyr::filter(tresults_4h, estimate > 1, p.adjusted < 0.05) %>%
              dplyr::arrange(desc(estimate))
de_Gat201up_4h

readr::write_tsv(de_Gat201up_4h, "~/GitHub/CryptoGat201RNASeq_draft_b/results_deseq/Gat201up_padj05_4h.txt")

readr::write_lines(de_Gat201up_4h$gene, "~/GitHub/CryptoGat201RNASeq_draft_b/results_deseq/Gat201up_padj05_4h_genesonly.csv")
                

de_Gat201down_4h <- dplyr::filter(tresults_4h, estimate < -1, p.adjusted < 0.05) %>%
                dplyr::arrange(estimate)
de_Gat201down_4h

readr::write_tsv(de_Gat201down_4h, "~/GitHub/CryptoGat201RNASeq_draft_b/results_deseq/Gat201down_padj05_4h.txt")

readr::write_lines(de_Gat201down_4h$gene, "~/GitHub/CryptoGat201RNASeq_draft_b/results_deseq/Gat201down_padj05_4h_genesonly.csv")

```

## Gat201 deletionv WT 4h Volcano Plot

```{r volcano 4h}

#Volcano plot with significant genes labelled

volcano_4h <- 
  ggplot(data = tresults_4h, aes(x=estimate,y=-log10(p.value))) +
         geom_point(colour = "grey50",size = 0.5) +
         geom_point(data = de_Gat201up_4h, colour = Gat201up_colour) +
         geom_point(data = de_Gat201down_4h, colour = Gat201down_colour) +
          geom_text_repel(aes(label=ifelse(estimate < -1 & -log10(p.value)> 1.3, (gene),'')), size= 2, max.overlaps = getOption("ggrepel.max.overlaps", default = 15))+
         geom_text_repel(aes(label=ifelse(estimate > 1 & -log10(p.value)> 1.3, (gene),'')), size = 2, max.overlaps = getOption("ggrepel.max.overlaps", default = 15))+
         scale_y_continuous(breaks = c(0,10,20,30,40,50,60,70,80,90,100),
                            limits = c(0,100), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        scale_x_continuous(breaks = seq(-6,6,2),
                            limits = c(-7,7), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        labs(x = "log2-fold change",
        title = "Gat201 mutants v WT at 4h")
volcano_4h

#volcano plot with Gat201 and possible downstream regulated genes

volcano4hsp <- 
  ggplot(data = tresults_4h, aes(x=estimate,y=-log10(p.value))) +
         geom_point(colour = "grey50",size = 0.5) +
         geom_point(data = de_Gat201up_4h, colour = Gat201up_colour) +
         geom_point(data = de_Gat201down_4h, colour = Gat201down_colour) +
   geom_text_repel(data = filter(tresults_4h, gene =="CNAG_01551"), label="Gat201",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_4h, gene =="CNAG_06762"), label="Gat204",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_4h, gene =="CNAG_063641"), label="Blp1",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_4h, gene =="CNAG_00883"), label="Ecm2201",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_4h, gene =="CNAG_05835"), label="Liv3",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_4h, gene =="CNAG_03012"), label="CQS1",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_4h, gene =="CNAG_01681"), label="FCY2",hjust = 1.1,vjust = -0.1, colour= "red")+
    geom_text_repel(data = filter(tresults_4h, gene =="CNAG_06401"), label="Ric8",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_4h, gene =="CNAG_07470"), label="Pde2",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_4h, gene =="CNAG_06241"), label="Cfo1",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_4h, gene =="CNAG_02777"), label="Pho84",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_4h, gene =="CNAG_04755"), label="Bck1",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_4h, gene =="CNAG_02357"), label="Mkk2",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_4h, gene =="CNAG_04514"), label="Mpk1",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_4h, gene =="CNAG_04969"), label="UGD1",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_4h, gene =="CNAG_01626"), label="Ada2",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_4h, gene =="CNAG_03024"), label="Rim15",hjust = 1.1,vjust = -0.1, colour= "black")+
    geom_text_repel(data = filter(tresults_4h, gene =="CNAG_04864"), label="Cir1",hjust = 1.1,vjust = -0.1, colour= "red")+
         scale_y_continuous(breaks = c(0,10,20,30,40,50,60,70,80,90,100),
                            limits = c(0,100), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        scale_x_continuous(breaks = seq(-5,5,1),
                            limits = c(-5,5), 
                            expand = c(0.01,0.01), 
                            oob = scales::squish) +
        labs(x = "log2-fold change",
             y = "-log10(p.value)",
             title = "Gat201 mutants v WT at 4h")
volcano4hsp

```


